src/lib/comms/sol-lwm2m.c

```diff
@@ -2853,31 +2900,62 @@ static uint8_t
 handle_delete(struct sol_lwm2m_client *client,
```

Comment:
You forgot to implement one thing here.
a request to delete '/' is only allowed during bootstrap